client/src/js/modules/contact/views/addcontact.js

```diff
@@ -17,26 +18,68 @@ define(['views/form',
         
         ui: {
             country: 'select[name=COUNTRY]',
+            pid: 'select[name=PERSONID]',
+            cn: 'input[name=CARDNAME]',
+            fn: 'input[name=FAMILYNAME]',
+            gn: 'input[name=GIVENNAME]',
+            pn: 'input[name=PHONENUMBER]',
+            ea: 'input[name=EMAILADDRESS]',
+            ln: 'input[name=LABNAME]',
+            ad: 'textarea[name=ADDRESS]',
+            ci: 'input[name=CITY]',
+            pc: 'input[name=POSTCODE]',
+        },
+
+        events: {
+            'change @ui.pid': 'fillUserInfo',
+        },
+
+        fillUserInfo: function() {
+            var u = this.users.findWhere({ PERSONID: this.ui.pid.val() })
+            if (u){
+                this.ui.cn.val(u.get('FULLNAME'))
+                this.ui.fn.val(u.get('FAMILYNAME'))
+                this.ui.gn.val(u.get('GIVENNAME'))
+                this.ui.pn.val(u.get('PHONENUMBER'))
+                this.ui.ea.val(u.get('EMAILADDRESS'))
+                this.ui.ln.val(u.get('LABNAME'))
+                this.ui.ad.val(u.get('ADDRESS'))
+                this.ui.ci.val(u.get('CITY'))
+                this.ui.pc.val(u.get('POSTCODE'))
+                this.ui.country.val(u.get('COUNTRY'))
+            }
         },
 
         initialize: function() {
             this.countries = new Countries()
             this.countries.state.pageSize = 9999
             this.ready = this.countries.fetch()
+            this.users = new Users()
+            this.users.queryParams.login = 1
+            this.ready2 = this.users.fetch()
         },
 
         createModel: function() {
             this.model = new Contact()
         },
-        
+
         onRender: function() {
             $.when(this.ready).done(this.populateCountries.bind(this))
+            $.when(this.ready2).done(this.populateUsers.bind(this))
```

Comment:
```suggestion
            Promise.all([this.countries.fetch(), this.users.fetch()]).then(() => {
                this.populateCountries();
                this.populateUsers();
            })
```

With `Promise.all`, we can do concurrent fetches without having to resort to bound functions or weird jQuery shenanigans, since the lexical scope of the anonymous function provided in the body of the thenable is always going to be the state of the global `this` object at that given moment in time